tdms/tests/system/read_config.py

```diff
@@ -0,0 +1,152 @@
+from pathlib import Path
+
+import yaml
+from utils import run_tdms
+
+# config files must provide this information
+required_fields = ["test_id", "tests"]
+# any runs must provide this information
+run_required_info = ["input_file", "reference"]
+# runs may also specify this information, but if they don't we use these defaults
+run_optional_info = {"fdtd_solver": False, "cubic_interpolation": False}
+
+
+def open_and_validate_config(config_file_path) -> dict:
+    # open the config file
+    config_file = open(config_file_path, "r")
+    # read data into dictionary
+    config = yaml.safe_load(config_file)
+    # can safely close config file after reading data
+    config_file.close()
+
+    for required_field in required_fields:
+        if required_field not in config.keys():
+            raise RuntimeError(f"{required_field} not provided in {config_file_path}")
+
+    # pass the data from the file back
+    return config
+
+
+def validate_run_information(runs: dict) -> None:
+    for test_id, test_info in runs.items():
+        for required_info in run_required_info:
+            if required_info not in test_info.keys():
+                raise RuntimeError(f"{required_info} not provided in {test_id}")
+    return
+
+
+class TDMSSystemTest:
```

Comment:
Can haz docstring plz?

Reply:
_Might_ have made a bit of a boilerplate code saving by making this a [dataclass](https://docs.python.org/3/library/dataclasses.html) and writing a `__post_init__` to do the validation and throwing runtime errors.

Probably not worth changing now.

Reply:
I have a local commit that adds docstrings to everything... just don't want to push it yet since since that'll make the tests restart and I want them to finish so they cache the data files :sweat_smile: